rollup/cmd/proposer_tool/app/app.go

```diff
@@ -48,122 +38,37 @@ func init() {
-			log.Crit("failed to close db connection", "error", err)
-		}
-	}()
-
-	// Init dbForReplay connection
-	dbForReplay, err := database.InitDB(cfg.DBConfigForReplay)
-	if err != nil {
-		log.Crit("failed to init dbForReplay connection", "err", err)
-	}
-	defer func() {
-		cancel()
-		if err = database.CloseDB(dbForReplay); err != nil {
-			log.Crit("failed to close dbForReplay connection", "error", err)
-		}
-	}()
-
-	// Init l2geth connection
-	l2Client, err := ethclient.Dial(cfg.L2Config.Endpoint)
-	if err != nil {
-		log.Crit("failed to connect l2 geth", "config file", cfgFile, "error", err)
-	}
 
 	startL2BlockHeight := ctx.Uint64(utils.StartL2BlockFlag.Name)
-	startL2Block, err := l2Client.BlockByNumber(context.Background(), big.NewInt(int64(startL2BlockHeight)))
-	if err != nil {
-		log.Crit("failed to get start l2 block", "startL2BlockHeight", startL2BlockHeight, "error", err)
-	}
-
-	chunk := &encoding.Chunk{Blocks: []*encoding.Block{{Header: startL2Block.Header()}}}
-
-	prevChunk, err := orm.NewChunk(dbForReplay).GetParentChunkByBlockNumber(subCtx, startL2BlockHeight)
-	if err != nil {
-		log.Crit("failed to get previous chunk", "error", err)
-	}
-
-	var startQueueIndex uint64
-	if prevChunk != nil {
-		startQueueIndex = prevChunk.TotalL1MessagesPoppedBefore + prevChunk.TotalL1MessagesPoppedInChunk
-	}
-
-	for _, tx := range startL2Block.Transactions() {
-		if tx.Type() == gethTypes.L1MessageTxType {
-			startQueueIndex++
-		}
-	}
-
-	_, err = orm.NewChunk(db).InsertTestChunkForProposerTool(subCtx, chunk, encoding.CodecV0, startQueueIndex)
-	if err != nil {
-		log.Crit("failed to insert chunk", "error", err)
-	}
-
-	batch := &encoding.Batch{
-		Index:                      0,
-		TotalL1MessagePoppedBefore: 0,
-		ParentBatchHash:            common.Hash{},
-		Chunks:                     []*encoding.Chunk{chunk},
-	}
-
-	var dbBatch *orm.Batch
-	dbBatch, err = orm.NewBatch(db).InsertBatch(subCtx, batch, encoding.CodecV0, rutils.BatchMetrics{})
-	if err != nil {
-		log.Crit("failed to insert batch", "error", err)
-	}
-
-	if err = orm.NewChunk(db).UpdateBatchHashInRange(subCtx, 0, 0, dbBatch.Hash); err != nil {
-		log.Crit("failed to update batch hash for chunks", "error", err)
-	}
-
-	registry := prometheus.DefaultRegisterer
 
 	genesisPath := ctx.String(utils.Genesis.Name)
 	genesis, err := utils.ReadGenesis(genesisPath)
 	if err != nil {
 		log.Crit("failed to read genesis", "genesis file", genesisPath, "error", err)
 	}
 
+	minCodecVersion := encoding.CodecVersion(ctx.Uint(utils.MinCodecVersionFlag.Name))
+
 	// sanity check config
 	if cfg.L2Config.BatchProposerConfig.MaxChunksPerBatch <= 0 {
 		log.Crit("cfg.L2Config.BatchProposerConfig.MaxChunksPerBatch must be greater than 0")
 	}
 	if cfg.L2Config.ChunkProposerConfig.MaxL2GasPerChunk <= 0 {
-		log.Crit("cfg.L2Config.ChunkProposerConfig.MaxL2GasPerChunk must be greater than 0")
+		log.Crit("cfg.L2Config.BatchProposerConfig.MaxChunksPerBatch must be greater than 0")
```

Comment:
wrong error message

Reply:
Oh yes. fixed in [6e7bc38](https://github.com/scroll-tech/scroll/pull/1645/commits/6e7bc388e1dc5ffc291ae0e7a3c5c5c92d7056aa).